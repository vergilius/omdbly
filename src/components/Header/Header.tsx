import React from 'react';
import styled from 'styled-components';
import {Flex} from '../Layout';

const HeaderWrapper = styled(Flex)`
  position: fixed;
  top: 0;
  left: 0;
  width: 100vw;
  height: ${({theme}) => `${theme.size.xxxlarge}`};
  background: ${({theme}) => `${theme.colors.red}`};
  padding: ${({theme}) => `${theme.size.default}`};
`;
const HeaderContainer = styled(Flex)`
  max-width: 1024px;
  width: 100%;
  margin: 0 auto;
`;

const Header = ({children}) => (
  <HeaderWrapper justifyContent="center">
    <HeaderContainer>
      {children}
    </HeaderContainer>
  </HeaderWrapper>
)

export default Header;
