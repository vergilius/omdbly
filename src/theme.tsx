const breakpoints = {
  large: '768px',
};

export interface Theme {
  breakpoints: {
    large: string
  },
  mediaQueries: {
    large: string
  },
  size: {
    none: string,
    xxsmall: string,
    xsmall: string,
    small: string,
    default: string,
    medium: string,
    large: string,
    xlarge: string,
    xxlarge: string,
    xxxlarge: string,
  },
  fontFamily: {
    logo: string,
    body: string,
  },
  fontSizes: {
    logo: number,
    body: number,
    headline: number,
  },
  colors: {
    light: string,
    dark: string,
    darkAlt: string,
    blue: string,
    red: string,
  }
};

const theme: Theme = {
  breakpoints,
  mediaQueries: {
    large: `@media screen and (min-width: ${breakpoints.large})`,
  },
  size: {
    none: '0',
    xxsmall: '4px',
    xsmall: '8px',
    small: '10px',
    default: '16px',
    medium: '24px',
    large: '32px',
    xlarge: '48px',
    xxlarge: '64px',
    xxxlarge: '96px',
  },
  fontFamily: {
    logo: '\'Limelight\', cursive',
    body: '\'Open Sans\', sans-serif',
  },
  fontSizes: {
    body: 16,
    headline: 20,
    logo: 64
  },
  colors: {
    light: '#ffffff',
    dark: '#4f506e',
    darkAlt: '#393c54',
    blue: '#1ab3fb',
    red: '#fb5464',
  }
};

export default theme;
