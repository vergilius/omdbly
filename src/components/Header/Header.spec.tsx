import React from 'react';
import styled, {ThemeProvider} from 'styled-components';
import Header from './Header';
import {mount} from 'enzyme';
import theme from '../../theme';

describe('<Header />', () => {
  it('renders properly children components', () => {
    const Dummy = () => <div>dummy component</div>;
    const wrapper = mount(
      <ThemeProvider theme={theme}>
        <Header>
          <Dummy />
        </Header>
      </ThemeProvider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
