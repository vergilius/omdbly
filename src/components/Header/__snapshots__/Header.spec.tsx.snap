// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders properly children components 1`] = `ReactWrapper {}`;
